src/main/java/uk/gov/hmcts/reform/managecase/repository/DefaultDefinitionStoreRepository.java

```diff
@@ -21,13 +24,23 @@ public DefaultDefinitionStoreRepository(DefinitionStoreApiClient definitionStore
     }
 
     @Override
+    @Cacheable(value = "challengeQuestions",
+        condition = "#root.target.getLatestVersion(#caseTypeId) == #root.target.versions[#caseTypeId]")
```

Comment:
I guess cache condition/unless are for evaluating whether result should cache or not 🤔  
May be worth to add some cache IT tests for this class?